scripts/utils/verify-scripts.js

```diff
@@ -0,0 +1,115 @@
+    globalPriceStorage = {},
+    logActivated = false
+  ) {
+    const log = logActivated ? (...a) => console.log(...a) : () => {}
+
+    const BatchExchangeArtifact = require("@gnosis.pm/dex-contracts/build/contracts/BatchExchange")
+    const networkId = await web3.eth.net.getId()
+    const BatchExchange = new web3.eth.Contract(BatchExchangeArtifact.abi, BatchExchangeArtifact.networks[networkId].address)
+
+    const auctionElementsDecoded = await getOrdersPaginated(BatchExchange, pageSize)
+    const bracketTraderAddresses = brackets.map(address => address.toLowerCase())
+
+    const GnosisSafe = artifacts.require("GnosisSafe")
+    const gnosisSafe = await GnosisSafe.deployed()
+
+    // Fetch all token infos(decimals, symbols etc) and prices upfront for the following verification
+    const relevantOrders = auctionElementsDecoded.filter(order => bracketTraderAddresses.includes(order.user.toLowerCase()))
+    const BatchExchangeContract = Contract(require("@gnosis.pm/dex-contracts/build/contracts/BatchExchange"))
+    BatchExchangeContract.setNetwork(web3.network_id)
+    BatchExchangeContract.setProvider(web3.currentProvider)
+    const exchange = await BatchExchangeContract.deployed()
+
+    const tradedTokenIds = new Set()
+    for (const order of relevantOrders) {
+      tradedTokenIds.add(order.sellToken)
+      tradedTokenIds.add(order.buyToken)
+    }
+    const tokenInfo = fetchTokenInfoFromExchange(exchange, Array.from(tradedTokenIds))
+
+    for (const order of relevantOrders) {
+      await getDexagPrice(
+        (await tokenInfo[order.sellToken]).symbol,
+        (await tokenInfo[order.buyToken]).symbol,
+        globalPriceStorage
+      )
+      await getDexagPrice((await tokenInfo[order.sellToken]).symbol, "USDC", globalPriceStorage)
+    }
+
+    log("1. Verify that the owner of the brackets is the masterSafe")
+    await Promise.all(
+      bracketTraderAddresses.map(async bracketTrader => {
+        assert(await isOnlySafeOwner(masterSafe, bracketTrader, artifacts), "Owners are not set correctly")
+      })
+    )
+
+    log("2. Verify that masterCopy of brackets is the known masterCopy")
+    await Promise.all(
+      bracketTraderAddresses.map(async bracketTrader => {
+        assert.equal(
+          (await getMasterCopy(bracketTrader)).toString().toLowerCase(),
+          gnosisSafe.address.toString().toLowerCase(),
+          "MasterCopy not set correctly"
+        )
+      })
+    )
+
+    log("3. Verify that each bracket has only two orders")
+    await Promise.all(
+      bracketTraderAddresses.map(async bracketTrader => {
+        const ownedOrders = relevantOrders.filter(order => order.user.toLowerCase() == bracketTrader)
+        assert(ownedOrders.length == 2, "order length is not correct")
+      })
+    )
+
+    log("4. Verify that each bracket can not loose tokens by selling and buying consecutively via their two orders")
+    await Promise.all(
+      bracketTraderAddresses.map(async bracketTrader => {
```

Comment:
The following block is not async, so a for loop should be enough.

Reply:
Good catch!